src/lib.rs

```diff
@@ -526,13 +691,15 @@ mod tests {
     // produce a recursive SNARK
     let res = RecursiveSNARK::prove(
       &pp,
-      num_steps,
+      Some(num_steps),
       <G1 as Group>::Scalar::one(),
       <G2 as Group>::Scalar::zero(),
     );
     assert!(res.is_ok());
     let recursive_snark = res.unwrap();
 
+    assert_eq!(num_steps, recursive_snark.num_steps);
```

Comment:
Is this assertion a sanity check or required for verifying that the prover faithfully executed the specified number of steps? If it is the latter, shouldn't this automatically be checked by the verify method?

Reply:
It was a sanity check, but you are right that since it is now included in `RecursiveSNARK`, we should fail verification if it doesn't match what is supplied. I added that logic to the verifier with some minimal negatives tests that `num_steps` must be correct.